src/libs/tools/src/backendbuilder.cpp

```diff
@@ -409,6 +415,40 @@ void BackendBuilder::addPlugin (PluginSpec const & plugin)
 		newPlugin.appendConfig (provides.getConfig ());
 	}
 
+	// call plugin's checkconf function (if provided)
+	// this enables a plugin to verify its configuration at mount time
+	checkConfPtr checkConfFunction = nullptr;
+	try
+	{
+		checkConfFunction = reinterpret_cast<checkConfPtr> (pluginDatabase->getSymbol (newPlugin, "checkconf"));
+	}
+	catch (...)
+	{
+		// the checkconf operation is optional, so no worries if it was not found
+	}
+
+	if (checkConfFunction != nullptr)
+	{
+		ckdb::Key * errorKey = ckdb::keyNew (0);
+		ckdb::KeySet * pluginConfig = newPlugin.getConfig ().dup ();
+		int checkResult = checkConfFunction (errorKey, pluginConfig);
+		if (checkResult == -1)
+		{
+			ckdb::ksDel (pluginConfig);
+			throw PluginConfigInvalid (errorKey);
+		}
+		else if (checkResult == 1)
+		{
+			KeySet modifiedPluginConfig = KeySet (pluginConfig);
+			newPlugin.appendConfig (modifiedPluginConfig);
```

Comment:
It's not possible to remove Keys from a KeySet. Should we be able to receive a whole new KeySet from the `checkconf` function? So the signature would change to:

    int checkconf (ckdb::Key * errorKey, ckdb::KeySet ** config)